### PR TITLE
Tobi/add stable token data

### PR DESCRIPTION
**Pull request by ghost**
This adds stable token data for inflation and total supply. 

It is tested by running it locally for a while, the output looks good and can be seen in `celo-testnet-production.tobi_eksportisto.data_stable_token_test`

##### Open Question
How often should we fetch the stable token data? Every block? Once every hour, every epoch?

### CODE REVIEW

**Comment by pedro-clabs** [comment]
> How often should we fetch the stable token data? Every block? Once every hour, every epoch?

I'd be in favour of doing it as granular as possible. My suggestion would be every block (which your "return true" implementation of `ShouldCollect` already does).

File: indexer/proc_stableToken.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+package indexer
+
+import (
+	"context"
+
+	"github.com/celo-org/celo-blockchain/accounts/abi/bind"
+	"github.com/celo-org/celo-blockchain/log"
+	//"github.com/celo-org/eksportisto/utils"
```
Comment (pedro-clabs) [inline indexer/proc_stableToken.go:8]: If they're not being used, I'd suggest removing them.

File: indexer/proc_stableToken.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	logger       log.Logger
+	stableToken *contracts.StableToken
+}
+
+func (proc *stableTokenProcessor) EventHandler() (registry.ContractID, EventHandler) {
+	return "", nil
+}
+
+func (proc *stableTokenProcessor) ShouldCollect() bool {
+	// TODO: How often should it collect the data? Per Block? Per Epoch?
```
Comment (pedro-clabs) [inline indexer/proc_stableToken.go:47]: Please remove the TODO once you're sure that it'll be collecting once per block.

File: indexer/proc_stableTokenEUR.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+func (proc stableTokenEURProcessor) ObserveMetrics(ctx context.Context) error {
+	/*
+	opts := &bind.CallOpts{
+		BlockNumber: proc.blockNumber,
+		Context:     ctx,
+	}
+	*/
+	// no metrics for now
+	return nil
+}
```
Comment (pedro-clabs) [inline indexer/proc_stableTokenEUR.go:97]: I'd suggest handling `StableTokenEUR` on `proc_stableToken.go`. You could use the `InitProcessors` method to return more than one processor, each bound to a different stable token. You could use https://github.com/celo-org/eksportisto/blob/master/indexer/proc_celoToken.go as an example.

File: indexer/proc_stableToken.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	rows <- contractRow.ViewCall("totalSupply", "totalSupply", totalSupply.String())
+
+	// StableToken.GetInflationParameters()
+	inflationFactor, inflationRate, updatePeriod, factorLastUpdated, err := proc.stableTokenContract.GetInflationParameters(opts)
+	if err != nil {
+		return errors.Wrap(err, 0)
+	}
+	rows <- contractRow.ViewCall(
+		"getInflationParameters",
+		"inflationFactor",helpers.FromFixed(inflationFactor),
```
Comment (pedro-clabs) [inline indexer/proc_stableToken.go:105]: Nit/style: add a space after the comma for consistency.

File: indexer/proc_stableToken.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	}
+	return processors, nil
+}
+
+type stableTokenProcessor struct {
+	*blockHandler
+	logger           log.Logger
+	stableTokenContract    *contracts.StableToken
+	stableTokenRegistryID  registry.ContractID
+	totalSupplyGauge prometheus.Gauge
```
Comment (pedro-clabs) [inline indexer/proc_stableToken.go:71]: Nit/style: I think in other files we're keeping these aligned. Would you mind trying to run `go fmt`?

File: indexer/proc_stableToken.go (hunk 4 above)
Comment (tobikuhlmann) [reply indexer/proc_stableToken.go]: Yes, thanks

**Pull request merged**